src/App.tsx

```diff
@@ -1,19 +1,171 @@
 import React from 'react';
-import './App.css';
+import classNames from 'classnames';
+import { TodoList } from './components/TodoList';
+import './App.scss';
 
 import users from './api/users';
+import todos from './api/todos';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todosArr: Todo[];
+  todoTitleValue: string;
+  todoUserValue: string;
+  invalidUser: boolean;
+  invalidTitle: boolean;
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todosArr: [...todos],
+    todoTitleValue: '',
+    todoUserValue: '',
+    invalidUser: false,
+    invalidTitle: false,
+  };
+
+  todoUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      todoUserValue: value,
+      invalidUser: false,
+    });
+  };
+
+  todoTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      todoTitleValue: value,
+      invalidTitle: false,
+    });
+  };
+
+  onSubmitAction = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (this.state.todoUserValue && this.state.todoTitleValue.trim()) {
+      const arr = [...this.state.todosArr].sort((a, b) => a.id - b.id);
+      const newId = arr[arr.length - 1].id + 1;
```

Comment:
you can specify start id in a state and do not sort todos it every time. Just add 1 here